test/parallel_api/iterator/permutation_iterator.pass.cpp

```diff
@@ -39,13 +81,490 @@ main()
     size_t num_elements = std::distance(first, last) / 2 + std::distance(first, last) % 2;
     auto permutation_first = dpl::make_permutation_iterator(first, multiply_index_by_two());
     auto permutation_last = permutation_first + num_elements;
-    auto it = std::copy(TestUtils::default_dpcpp_policy, permutation_first, permutation_last, result.begin());
+
+    auto it = std::copy(dpl::execution::make_device_policy(q), permutation_first, permutation_last, result.begin());
     auto count = ::std::distance(result.begin(), it);
 
     for (int i = 0; i < count; i++)
         ::std::cout << result[i] << " ";
 
     ::std::cout << ::std::endl;
+}
+
+// test_sort : dpl::sort -> __parallel_stable_sort
+// Requirements: only for random_access_iterator
+DEFINE_TEST(test_sort)
+{
+    DEFINE_TEST_CONSTRUCTOR(test_sort)
+
+    template <typename TIterator>
+    void generate_data(TIterator itBegin, TIterator itEnd)
+    {
+        const auto testing_n = ::std::distance(itBegin, itEnd);
+
+        for (auto it = itBegin; it != itEnd; ++it)
+            *it = testing_n - std::distance(itBegin, it);
```

Comment:
probably it make sense to add just integer counter instead of distance calculation every iteration within the loop?

Reply:
Fixed